Auto email response

cmcewan93 opened the pull request.

Commit:
feat: updated apply/page/api route to send confirmation email on successful appplication submission

/apply/page/api triggers sendConfirmationEmail when an application has been validated and
succesfully submitted, submit application function updated to return isValid boolena and user email
when successful

osf-112

Commit:
feat: email template and mailer service created

Implemented the mailer service with handlebars used for email template, completed with styling

OSF-112

junminahn commented on app/services/mailer/mailer.js:
```diff
@@ -0,0 +1,56 @@
+
+async function sendConfirmationEmail(toEmail) {
+  // create reusable transporter object
+  let transporter = nodemailer.createTransport({
```
can we create a transport only one time to use for the express server instead of creating each time?

junminahn commented on app/services/mailer/views/confirmation.handlebars:
```diff
@@ -0,0 +1,50 @@
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>Launch Online Application</title>
+
+  <style>
```
can we avoid using embedded styles in email HTML templates?
https://content.myemma.com/blog/css-in-html-emails-what-you-need-to-know-to-get-started

junminahn commented on app/pages/api/apply/[page].js:
```diff
@@ -1,4 +1,5 @@
 import { LAST_PAGE, submitApplication, pageForward } from 'services/application';
+import sendConfirmationEmail from '../../../services/mailer/mailer';
```
```suggestion
import sendConfirmationEmail from 'services/mailer/mailer';
```

junminahn commented on app/pages/api/apply/[page].js:
```diff
@@ -19,9 +20,15 @@ async function handler(req, res) {
-    await submitApplication(context);
+    let success = await submitApplication(context);
+    if (success !== undefined) {
+      let [result, res, userEmail] = success;
```
we have `req` object above `const context = { req, newData: newFormData, page, js };`
and can find `res` object within the `req` objects as well.
so I believe we might not need to change the return values in `submitApplication` function.

junminahn reviewed (commented):
It looks great! and I left couple of minor comments.

Commit:
refactor: removed embedded styling

replaced embedded styling in html template with inline styling

OSF-112

Commit:
refactor: added config folder for mailer and minor syntax changes

moved transport object, mail options and view options for handlebar into a config folder

osf-112
